Type error when converting results to polars

Hi, I have encountered an issue while building a project using the latest version available on cargo, `0.9.0`, with the `polars` feature enabled. The error is as follows:
```rust
error[E0308]: mismatched types
 40 |         .infer_schema_len(Some(5))
    |                           ---- ^ expected `NonZero<usize>`, found integer
    |                           |
    |                           arguments to this enum variant are incorrect
    |
    = note: expected struct `NonZero<usize>`
```
The issue appears to be in line 40 of the file `polars.rs`.  My build environment was the `rust:slim-bookworm` docker image on ARM64 using rust version `1.79.0-aarch64-unknown-linux-gnu`.

A potential solution I suggest is as follows, starting from line 39:
```rust
let df = JsonReader::new(reader)
    .with_json_format(JsonFormat::Json)
    .infer_schema_len(Some(NonZeroUsize::new(5).unwrap()))
    .finish()?;
```
The modification above resolves the issue on my end.

--- a/snowflake-api/Cargo.toml
+++ b/snowflake-api/Cargo.toml
@@ -41,8 +41,8 @@ url = "2"
 uuid = { version = "1", features = ["v4"] }
 
 # polars-support
-polars-core = { version = ">=0.32", optional = true }
-polars-io = { version = ">=0.32", features = [
+polars-core = { version = ">=0.41", optional = true }
+polars-io = { version = ">=0.41", features = [
     "json",
     "ipc_streaming",
 ], optional = true }

--- a/snowflake-api/src/polars.rs
+++ b/snowflake-api/src/polars.rs
@@ -1,5 +1,5 @@
 use std::convert::TryFrom;
-
+use std::num::NonZeroUsize;
 use bytes::{Buf, Bytes};
 use polars_core::frame::DataFrame;
 use polars_io::ipc::IpcStreamReader;
@@ -37,7 +37,7 @@ fn dataframe_from_json(json_result: &JsonResult) -> Result<DataFrame, PolarsCast
     let reader = std::io::Cursor::new(json_string.as_bytes());
     let df = JsonReader::new(reader)
         .with_json_format(JsonFormat::Json)
-        .infer_schema_len(Some(5))
+        .infer_schema_len(NonZeroUsize::new(5))
         .finish()?;
     Ok(df)
 }

--- a/snowflake-api/src/polars.rs
+++ b/snowflake-api/src/polars.rs
@@ -1,5 +1,6 @@
 use std::convert::TryFrom;
 use std::num::NonZeroUsize;
+
 use bytes::{Buf, Bytes};
 use polars_core::frame::DataFrame;
 use polars_io::ipc::IpcStreamReader;

